src/main/kotlin/org/rust/lang/core/resolve/scope/RustResolveScope.kt

```diff
@@ -1,9 +1,256 @@
 package org.rust.lang.core.resolve.scope
 
-import org.rust.lang.core.psi.RustCompositeElement
-import org.rust.lang.core.psi.RustNamedElement
+import com.intellij.openapi.module.Module
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.PsiUtilCore
+import org.rust.cargo.util.AutoInjectedCrates
+import org.rust.cargo.util.cargoProject
+import org.rust.cargo.util.getPsiFor
+import org.rust.cargo.util.preludeModule
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustFile
+import org.rust.lang.core.psi.impl.mixin.isStarImport
+import org.rust.lang.core.psi.impl.rustMod
+import org.rust.lang.core.psi.util.module
 
 interface RustResolveScope : RustCompositeElement {
-    val declarations: Collection<RustNamedElement>
+    data class Context(
```

Comment:
`Entry` is definitely an integral part of the `ResolveScope`.  `ResolveScope` is the thing which contains `Entries`, and `Entries` are more then just `NamedElements`, because they include aliasing info and provide laziness. 

As for the `Context`, it should be simply made `private` to this file (done). 

Please note that this implementation of `ResolveScope` knows nothing about `RRE` which is a nice property which I'd love to preserve.